manifestProvider.go

```diff
@@ -0,0 +1,30 @@
+package onfleet
+
+type TurnByTurn struct {
+	StartAddress    string   `json:"startAddress"`
+	EndAddress      string   `json:"endAddress"`
+	ETA             int64    `json:"eta"`
+	DrivingDistance string   `json:"drivingDistance"`
+	Steps           []string `json:"steps"`
+}
+
+type Driver struct {
+	Name  string `json:"name"`
+	Phone string `json:"phone"`
+}
+
+type ManifestGenerateParams struct {
+	WorkerId string `json:"workerId"`
+	HubId    string `json:"hubId"`
+}
+
+type DeliveryManifest struct {
+	ManifestDate  int64              `json:"manifestDate"`
+	DepartureTime int64              `json:"departureTime"`
+	Driver        Driver             `json:"driver"`
+	Vehicle       WorkerVehicleParam `json:"vehicle"`
+	HubAddress    string             `json:"hubAddress"`
+	Tasks         []Task             `json:"tasks"`
+	TurnByTurn    []TurnByTurn       `json:"turnByTurn"`
+	TotalDistance string             `json:"totalDistance"`
+}
```

Comment:
i am unfamiliar with delivery manifest schema and which (if any) fields may be `null` to indicate absence. but something to consider is that Go data types have default values so let's say `TurnByTurn` field `ETA` should actually be `null` when not present ... the way the code is currently `ETA` would default to `0` when absent. all pointers have default value of `nil` in Go so you could modify (wherever necessary) to be like the following example
```
ETA *int64 `json:"eta"`
```

Reply:
Thanks for pointing that out, I didn't know that. I made sure to check all cases and as far as I'm concerned this code should work fine.